src/ucp/wireup/wireup.c

```diff
@@ -1803,5 +1803,44 @@ unsigned ucp_ep_init_flags(const ucp_worker_h worker,
     return flags;
 }
 
+double ucp_wireup_iface_lat_distance_v1(ucp_context_h context,
+                                        const ucs_linear_func_t *latency,
+                                        const ucs_sys_dev_distance_t *distance)
+{
+    return (context->config.ext.proto_enable) ?
+        (latency->c + distance->latency) : latency->c;
+}
+
+double ucp_wireup_iface_lat_distance_v2(ucp_context_h context,
+                                        const ucs_linear_func_t *latency,
+                                        const ucs_sys_dev_distance_t *distance)
+{
+    ucs_linear_func_t lat;
+
+    if (context->config.ext.proto_enable)
+    {
+        lat    = *latency;
+        lat.c += distance->latency;
+        return ucp_tl_iface_latency(context, &lat);
+    } else {
+        return ucp_tl_iface_latency(context, latency);
+    }
+}
+
+double ucp_wireup_iface_bw_distance(ucp_context_h context,
```

Comment:
maybe pass worker instead of context to reduce some code when this function is called?